tests/lib/config/flat-config-array.js

```diff
@@ -1987,4 +1987,68 @@ describe("FlatConfigArray", () => {
         });
 
     });
+
+    // https://github.com/eslint/eslint/issues/12592
+    describe("Shared references between rule configs", () => {
+
+        it("shared rule config should not cause a rule validation error", () => {
+
+            /* eslint-disable-next-line prefer-const -- need to allow mutation inside FlatConfigArray */
+            let ruleConfig = ["error", {}];
+
+            const configs = new FlatConfigArray([{
+                rules: {
+                    camelcase: ruleConfig,
+                    "default-case": ruleConfig
+                }
+            }]);
+
+            configs.normalizeSync();
+
+            const config = configs.getConfig("foo.js");
+
+            assert.deepStrictEqual(config.rules, {
+                camelcase: [2, {
+                    ignoreDestructuring: false,
+                    ignoreGlobals: false,
+                    ignoreImports: false
+                }],
+                "default-case": [2, {}]
+            });
+
+        });
+
+
+        it("should throw rule validation error for camelcase", async () => {
+
+
+            /* eslint-disable-next-line prefer-const -- need to allow mutation inside FlatConfigArray */
+            let ruleConfig = ["error", {}];
+
+            const configs = new FlatConfigArray([{
+                rules: {
+                    camelcase: ruleConfig,
+                    "default-case": ruleConfig,
+
+                    /* eslint-disable-next-line no-dupe-keys -- needed for testing */
+                    camelcase: [
+                        "error",
+                        {
+                            ignoreDestructuring: Date
+                        }
+
+                    ]
```

Comment:
```suggestion
                    "default-case": ruleConfig,

                    camelcase: [
                        "error",
                        {
                            ignoreDestructuring: Date
                        }

                    ]
```

I believe duplicate keys don't affect this testing, the previous one is just overwritten in the created object. Perhaps the configuration was intended to have two config objects? (though the test might be even better with one to make sure we're always cloning rule configs)

Reply:
I copied this from the tests that were originally submitted for fixing this bug. Adding a second config would make sense. :+1: